### PR TITLE
Explicitly discard result of std::remove

### PR DESCRIPTION
**Pull request by dneto0**
Fixes Android arm-64-v8a build with NDK r14.

### CODE REVIEW

File: source/opt/decoration_manager.cpp [hunk 1]
```diff
@@ -305,12 +305,13 @@ void DecorationManager::RemoveInstructionFromTarget(ir::Instruction* inst,
                                                     const uint32_t target_id) {
   auto const group_iter = group_to_decoration_insts_.find(target_id);
   if (group_iter != group_to_decoration_insts_.end()) {
-    remove(group_iter->second.begin(), group_iter->second.end(), inst);
+    (void)std::remove(group_iter->second.begin(), group_iter->second.end(),
+                      inst);
```
Comment (atgoo) [inline source/opt/decoration_manager.cpp]: Consider
group_iter->second.erase(std::remove(group_iter->second.begin(), group_iter->second.end(), inst), group_iter->second.end());

Comment (dneto0) [reply source/opt/decoration_manager.cpp]: That physically resizes the array, which seems unnecessary.  I don't have strong intuition about whether that's significantly wasteful.  I'll make the change (logically, anyway).

Comment (dneto0) [reply source/opt/decoration_manager.cpp]: Vector, not array.

**Comment by dneto0** [comment]
Rebased against master, and rewrote the patch to erase the removed elements.

**Comment by dneto0** [comment]
Please take another look.

**Comment by dneto0** [comment]
Rebased and pushed into master as 188cd3780d76

**Pull request closed without merging**